src/ClientData/ScopeIdProvider.cpp

```diff
@@ -35,17 +36,24 @@ std::unique_ptr<NameEqualityScopeIdProvider> NameEqualityScopeIdProvider::Create
 
   absl::flat_hash_map<ScopeId, const ScopeInfo> scope_id_to_info;
   absl::flat_hash_map<const ScopeInfo, ScopeId> scope_info_to_id;
+  absl::flat_hash_map<ScopeId, const FunctionInfo> scope_id_to_function_info;
 
   for (const auto& instrumented_function : instrumented_functions) {
     const ScopeId scope_id{instrumented_function.function_id()};
     const ScopeInfo scope_info(instrumented_function.function_name(),
                                ScopeType::kDynamicallyInstrumentedFunction);
     scope_id_to_info.emplace(scope_id, scope_info);
     scope_info_to_id.emplace(scope_info, scope_id);
+    scope_id_to_function_info.emplace(
+        scope_id,
+        FunctionInfo{instrumented_function.file_path(), instrumented_function.file_build_id(),
```

Comment:
Maybe use `std::piecewise_construct`?  Might hurt readability though. Other reviewer's opinion?

Reply:
https://en.cppreference.com/w/cpp/container/map/try_emplace afaict try_emplace already does this so I can't use try_emplace&piecewise_construct together?

Reply:
Oh, didn't know that. But this makes it even better. You can write 

```
scope_id_to_function_info.try_emplace(
        scope_id, instrumented_function.file_path(), instrumented_function.file_build_id(),
        instrumented_function.function_virtual_address(), instrumented_function.function_size(),
        instrumented_function.function_name());
```
This way we instantiate `FunctionInfo` only once. 
But then again, I am not sure if it's readable enough. I wonder what you and @antonrohr think.

Reply:
Ah makes sense, thanks.  What do you think of a comment like this for readability?

Reply:
https://github.com/google/orbit/pull/4208#discussion_r974145391